vm/src/bench.rs

```diff
@@ -80,6 +80,28 @@ where
             .lookup_id(addr)
             .map_err(|e| anyhow!("failed to resolve address {}: {}", addr, e.to_string()))
     }
+
+    fn set_epoch(&mut self, epoch: ChainEpoch) {
+        replace_with::replace_with_or_abort(&mut self.executor, |e| {
+            let mut machine = e.into_machine().unwrap();
+            let engine_conf = (&machine.context().network).into();
+            let mut machine_ctx = machine.context().clone();
+            machine_ctx.epoch = epoch;
+            machine_ctx.initial_state_root = machine.flush().unwrap();
+            let machine = DefaultMachine::new(
+                &machine_ctx,
+                machine.into_store().into_inner(),
+                FakeExterns::new(),
```

Comment:
Here, there is currently no way to get the externs out of the machine. `Machine::externs(&self)` does exist but since the above line `machine.into_store()` takes ownership of the machine we cannot borrow it again. To prototype the functionality here, I've just replaced the generic parameter E with concrete type of FakeExterns. 

Alternatives here that would allow us to keep the generic flexibility over externs.
- add a function to Machine to allow a single function that takes ownership and returns a tuple of blockstore, externs
- add a function to Machine that allows explicit mutation of the `MachineContext`. Though this seems like a bit of an anti-pattern. My understanding is that the Machine shouldn't really mutate but rather new machines should be instantiated per epoch.  But maybe this is ok.

Reply:
Thanks, please move this comment into the code itself. We can also expand the functionality of FakeExterns to itself just be an indirection to user-provided implementations.